Code/Tools/ProjectManager/Source/DownloadController.h

```diff
@@ -57,16 +70,16 @@ namespace O3DE::ProjectManager
         void HandleResults(const QString& result, const QString& detailedError);
 
     signals:
-        void StartGemDownload(const QString& gemName, bool downloadNow);
-        void Done(const QString& gemName, bool success = true);
-        void GemDownloadAdded(const QString& gemName);
-        void GemDownloadRemoved(const QString& gemName);
-        void GemDownloadProgress(const QString& gemName, int bytesDownloaded, int totalBytes);
+        void StartObjectDownload(const QString& objectName, DownloadObjectType objectType, bool downloadNow);
+        void Done(const QString& objectName, bool success = true);
+        void ObjectDownloadAdded(const QString& objectName, DownloadObjectType objectType);
+        void ObjectDownloadRemoved(const QString& objectName, DownloadObjectType objectType);
+        void ObjectDownloadProgress(const QString& objectName, DownloadObjectType objectType, int bytesDownloaded, int totalBytes);
 
     private:
         DownloadWorker* m_worker;
         QThread m_workerThread;
-        QWidget* m_parent;
-        AZStd::vector<QString> m_gemNames;
+        AZStd::vector<DownloadableObject> m_objects;
```

Comment:
Because there are a lot of erasing occurring within this container, especially from the front.
A deque is a more suitable container, as it doesn't have the O(n) characters of erasing eleemnts from the front.